Test suite maintenance

aitbw opened the pull request:
This PR introduces several changes to this extension to get it to a working state; please, have a look at the commits attached, as they serve as a changelog.

This patch is still two specs short to be fully completed, but those tests **pass** when tested manually, hence the `WIP` and `RFC` tags on the title. I'd love to hear your opinions whether we should skip those tests for the sake of having this extension fixed or if we should invest more time on them.

aitbw committed:
Remove unnecessary before install scripts on Travis config

Before install scripts were causing the build to fail

aitbw committed:
Lock FactoryBot to v4.10 for Solidus versions lower than v2.5

aitbw committed:
Fix all deprecations warnings

aitbw committed:
Add missing I18n keys

aitbw committed:
Fix broken admin/volume_price_models#new view

aitbw committed:
Replace FactoryGirl -> FactoryBot

aitbw committed:
Specify Rails version for DB migrations

aitbw committed:
Fix incorrect handling of ENV variable to install a DB adapter

aitbw committed:
Update Hound configuration

aitbw committed:
Do not use static attributes on factories

aitbw committed:
Fix Git insecure data fetching warning

aitbw committed:
Fix failing admin/variants controller spec

aitbw committed:
Clean markup on views

aitbw committed:
Update Travis build matrix

It introduces the following changes:

* Add Solidus v2.8
* Remove Solidus v2.3

aitbw committed:
Lock SQLite3 to version 1.3

SQLite3 v1.4 was released on February 4th, which breaks compatibility
with ActiveRecord's adapter for said DB engine.

mdesantis commented on app/views/spree/admin/variants/_edit_fields.html.erb:
```diff
@@ -1,21 +1,14 @@
 <table class="table">
   <thead>
     <tr>
-      <th><%= Spree.t(:name) %></th>
```
I think you should let translation keys being symbols, you can achieve that prepending `:`: `:'spree.name'`. Symbols are better for keys usage, also from a performance point of view:

```ruby
require 'benchmark'

Benchmark.bmbm do |x|
  x.report('string') { 100_000_000.times { 'spree.create' } }
  x.report('symbol') { 100_000_000.times { :'spree.create' } }
end
```

```
Rehearsal ------------------------------------------
string   5.953468   0.000366   5.953834 (  5.953751)
symbol   3.605581   0.000002   3.605583 (  3.605522)
--------------------------------- total: 9.559417sec

             user     system      total        real
string   5.605612   0.000000   5.605612 (  5.605510)
symbol   3.213020   0.000000   3.213020 (  3.212970)
```

aitbw replied:
Very interesting @mdesantis :thinking:

Will fix as necessary, thanks!! :heart:

tvdeyen replied:
> I think you should let translation keys being symbols, you can achieve that prepending :: :'spree.name'. 

We do that all over the code base, but great point @mdesantis although I am not a big fan of the proposed change as this looks a bit weird.

What about (the even more correct):

```rb
t(:name, scope: :spree)
```

_:point_up: this is actually what `Spree.t` used to do for us behind the scenes._

mdesantis replied:
@tvdeyen yeah that's even better

mdesantis reviewed (commented):
Great job!!! Just one doubt

tvdeyen commented on app/views/spree/admin/variants/volume_prices.html.erb:
```diff
@@ -1,24 +1,24 @@
-    <%= button_link_to Spree.t(:add_volume_price), 'javascript:;', {
-      :'data-target' => 'tbody#volume_prices', class: 'btn-success spree_add_fields', id: 'add_volume_price'} %>
+    <%= link_to I18n.t('spree.add_volume_price'), 'javascript:;', {
+      :'data-target' => 'tbody#volume_prices', class: 'btn-success spree_add_fields', id: 'add_volume_price' } %>
```
We need to add the `btn` class as well otherwise this will not be displayed as button anymore (Solidus >= 2.4 uses Bootstrap classes for buttons). Also we should use the primary button color instead of a green success button, as this is the default button color in Solidus 

```suggestion
      :'data-target' => 'tbody#volume_prices', class: 'btn btn-primary spree_add_fields', id: 'add_volume_price' } %>
```

tvdeyen replied:
These changes can be tackled in a subsequent PR (that I already prepared locally). There are more admin UI changes necessary to update this extension to Solidus 2.4 and beyond.

tvdeyen commented:
@aitbw I converted this into a reviewable PR as I hoped that this will trigger the TravisCI cuild, but that didnt worked. Anyway this is very much appreciated work that I would like to see continued if any possible. Will send a review soon and try to run specs locally. Thanks :tada:

tvdeyen reviewed (commented):
Great work. I stil have three specs failing locally. Unfortunately the TravisCI build is not triggering. I think we need to close and re-open that PR. Please continue working on this :clap:

tvdeyen committed:
Fix the active record validation errors message

Activerecord has a build in way to translate validation messages. No
need to invent our own way.

tvdeyen commented:
@aitbw I pushed a fix for one of the failing specs and that also triggered a TravisCI build. I hope you don't mind :)

tvdeyen committed:
Do not install custom select2 handling

Solidus >= 2.4 does not use select2 for normal selectboxes anymore.

tvdeyen committed:
Use custom-select over select2

Solidus 2.4 stopped to use select2 for normal selects but use Bootstrap
custom-selects instead.

tvdeyen committed:
Add missing translation for "Total price"

tvdeyen committed:
Respond with head :ok to from delete volume prices

Rails 5.1 removed the `render nothing: true` and
recommends to use `head :ok` instead.

Works with all compatible Solidus and Rails versions.

tvdeyen reviewed (approved):
I added more fixes to made the specs pass. Everything else can be addressed in subsequent PRs. Thanks for this work! :tada:
